runtime/src/lib.rs

```diff
@@ -1039,13 +1111,55 @@ impl_runtime_apis! {
                     weights: stats.weights,
                 },
                 params: ModuleParams {
-                    name: params.name,
-                    address: params.address,
+                    name: String::from_utf8(params.name).expect("Name bytes should be valid utf8"),
+                    address: String::from_utf8(params.address).expect("Name bytes should be valid utf8"),
                     delegation_fee: params.delegation_fee,
                     controller: params.controller,
                 }
             }
         }
+
+        fn get_key_info(key: AccountId) -> KeyInfo {
+            let balance = SubspaceModule::get_balance_u64(&key);
+            let total_stake = SubspaceModule::get_global_stake_to(&key);
+            let mut stake_to: Vec<KeyStakeToInfo> = Vec::new();
+
+            for netuid in SubspaceModule::netuids() {
+                let  mut stake_to_module: Vec<(String, u64)> = Vec::new();
+
+                let stake_to_vec = SubspaceModule::get_stake_to_vector(netuid, &key);
+
+                if !stake_to_vec.is_empty() {
+                    for (module_key, amount) in stake_to_vec.iter() {
+                        let module_name = SubspaceModule::module_params(netuid, module_key).name;
+
+                        if *amount > 0 {
+                            stake_to_module.push((
+                                String::from_utf8(module_name).expect("Name bytes should be valid utf8"),
+                                *amount,
+                            ))
+                        }
+                    }
+                }
+
+                if !stake_to_module.is_empty() {
+                    let subnet_name_bytes = SubspaceModule::get_subnet_name(netuid);
+                    stake_to.push(
+                        KeyStakeToInfo {
+                            netuid,
+                            subnet_name: String::from_utf8(subnet_name_bytes).expect("Name bytes should be valid utf8"),
+                            stake_to_module
+                        }
+                    )
+                }
```

Comment:
You can use iterators here as well.

Reply:
How can I use iterators here?
This is only push action.
Can you show me example?

Reply:
Oh, I get what you are saying, my bad.